cmd/earthly/main.go

```diff
@@ -1271,24 +1294,31 @@ func (app *earthlyApp) before(context *cli.Context) error {
 	return nil
 }
 
-func (app *earthlyApp) configureSatellite(cc cloud.Client) error {
-	if !app.isUsingSatellite() || cc == nil {
+func (app *earthlyApp) configureSatellite(c *cli.Context, cc cloud.Client) error {
+	if c.IsSet("buildkit-host") && c.IsSet("satellite") {
+		return errors.New("cannot specify both buildkit-host and satellite")
+	}
+	if !app.isUsingSatellite(c) || cc == nil {
 		// If the app is not using a cloud client, or the command doesn't interact with the cloud (prune, bootstrap)
 		// then pretend its all good and use your regular configuration.
 		return nil
 	}
 
 	// When using a satellite, interactive and local do not work; as they are not SSL nor routable yet.
-	app.console.Warnf("Note: the Interactive Debugger, Interactive RUN commands, and Local Registries do not yet work on Earthly Satellites.")
+	app.console.Warnf("Note: the interactive debugger, interactive RUN commands, and fast output via embedded registry do not yet work on Earthly Satellites.")
 
 	// Set up extra settings needed for buildkit RPC metadata
-	app.satelliteName = app.cfg.Satellite.Name
-	app.satelliteOrg = app.cfg.Satellite.Org
+	if app.satelliteName == "" {
+		app.satelliteName = app.cfg.Satellite.Name
+	}
+	if app.satelliteOrg == "" {
+		app.satelliteOrg = app.cfg.Satellite.Org
+	}
```

Comment:
When would we encounter this new condition? It shouldn't be possible to land here without those being set? And if we do, and they are empty, setting them yields the same result?

Reply:
Cool. Initial thoughts was that this was for guards.

Man, we need to centralize/clean up this config override stuff and initialization stuff so its not spread out like this...